torch_geometric/nn/conv/message_passing.py

```diff
@@ -258,54 +265,68 @@ def propagate(self, edge_index: Adj, size: Size = None, **kwargs):
 
         # Otherwise, run both functions in separation.
         elif isinstance(edge_index, Tensor) or not self.fuse:
-            coll_dict = self.__collect__(self.__user_args__, edge_index, size,
-                                         kwargs)
-
-            msg_kwargs = self.inspector.distribute('message', coll_dict)
-            for hook in self._message_forward_pre_hooks.values():
-                res = hook(self, (msg_kwargs, ))
-                if res is not None:
-                    msg_kwargs = res[0] if isinstance(res, tuple) else res
-            out = self.message(**msg_kwargs)
-            for hook in self._message_forward_hooks.values():
-                res = hook(self, (msg_kwargs, ), out)
-                if res is not None:
-                    out = res
 
-            # For `GNNExplainer`, we require a separate message and aggregate
-            # procedure since this allows us to inject the `edge_mask` into the
-            # message passing computation scheme.
-            if self.__explain__:
-                edge_mask = self.__edge_mask__.sigmoid()
-                # Some ops add self-loops to `edge_index`. We need to do the
-                # same for `edge_mask` (but do not train those).
-                if out.size(self.node_dim) != edge_mask.size(0):
-                    edge_mask = edge_mask[self.__loop_mask__]
-                    loop = edge_mask.new_ones(size[0])
-                    edge_mask = torch.cat([edge_mask, loop], dim=0)
-                assert out.size(self.node_dim) == edge_mask.size(0)
```

Comment:
edge_mask needs to computed only once, when `i=0`.

Reply:
Yes. It's probably a good idea to set `decomposed_layers=1` in case self.__explain__=True`.

Reply:
@rusty1s @wsad1 I think your idea is very good, but I don’t know how to do it. Could you please help me modify it, or tell me how to do it.

Reply:
So you could add these lines in the `__init__`
```
if self.__explain__=True:
    self.decomposed_layers=1
```
And mention in the init docstring that this happens.

Reply:
@wsad1 Thank you for your help, I have modified it, do I resubmit the PR?

Reply:
You could make the modification to your forked [branch](https://github.com/ZhouAo-ZA/pytorch_geometric) and they will show up on this PR. I don't think we need to create a new PR for this.

Reply:
@wsad1 I have committed the code, thanks for the help.